src/tiled/scriptdialog.cpp

```diff
@@ -0,0 +1,330 @@
+    m_gridLayoutWidget->setGeometry(QRect(15, 15, minimumWidth()-30, minimumHeight()-30));
+    m_gridLayout = new QGridLayout(m_gridLayoutWidget);
+    m_gridLayout->setContentsMargins(15, 15, 15, 15);
+    m_gridLayout->setObjectName(QString::fromUtf8("m_gridLayout"));
+    // make the right-hand column more likely to stretch
+    m_gridLayout->setColumnStretch(0, m_leftColumnStretch);
+    m_gridLayout->setColumnStretch(1, m_rightColumnStretch);
+    m_rowIndex = 0;
+    addNewRow();
+}
+void ScriptDialog::clear()
+{
+    m_widgetNumber = 0;
+    QLayoutItem * item;
+    QLayout * sublayout;
+    QWidget * widget;
+    while ((item = m_gridLayout->takeAt(0))) {
+        if ((sublayout = item->layout()) != 0) {/* do the same for sublayout*/}
+        else if ((widget = item->widget()) != 0) {widget->hide(); delete widget;}
+        else {delete item;}
+    }
+    delete m_gridLayoutWidget;
+    initializeLayout();
+}
+void ScriptDialog::resize(const int width, const int height)
+{
+    QSize oldSize = size();
+    QDialog::resize(width, height);
+    QSize newSize = QSize(width, height);
+    m_gridLayout->setGeometry(QRect(15,15, width-30, height-30));
+    m_gridLayoutWidget->setGeometry(QRect(15,15, width-30, height-30));
+    setMinimumWidth(width);
+    setMinimumHeight(height);
+    adjustSize();
+    QResizeEvent event = QResizeEvent(newSize, oldSize);
+    resizeEvent(&event);
+}
+void ScriptDialog::resizeEvent(QResizeEvent* event)
+{
+    m_gridLayout->invalidate();
+    m_gridLayoutWidget->setGeometry(QRect(15,15, event->size().width()-30, event->size().height()-30));
+}
+QLabel *ScriptDialog::addLabel(const QString &text)
+{
+    return addLabel(text, false);
+}
+QLabel *ScriptDialog::addLabel(const QString &text, bool maxWidth)
+{
+    QLabel * label;
+    addNewRow();
+    checkIfSameType(typeid(label).name());
+    label = newLabel(text);
+    if (m_widgetsInRow == 0) {
+        m_gridLayout->addWidget(label, m_rowIndex, m_widgetsInRow, 1, maxWidth ? -1 : 1); // max width
+        QSizePolicy sizePolicy(QSizePolicy::Expanding, QSizePolicy::Fixed);
+        label->setSizePolicy(sizePolicy);
+        m_widgetsInRow++;
+        if (maxWidth){
+            label->setWordWrap(true);
+            addNewRow();
+        }
+    }
+    else
+        addDialogWidget(label);
+    return label;
+}
+Tiled::ScriptDialogWidget *ScriptDialog::addSeparator()
+{
+    return addSeparator(QString());
+}
+Tiled::ScriptDialogWidget *ScriptDialog::addSeparator(const QString &labelText)
+{
+    QFrame *line;
+    addNewRow();
+    m_gridLayout->addLayout(m_rowLayout, m_rowIndex,0, 1, -1); // span entire
+    QLabel * separatorLabel;
+    if(!labelText.isEmpty()){
+        separatorLabel= newLabel(labelText);
+        separatorLabel->setWordWrap(false);
+        m_rowLayout->addWidget(separatorLabel, 1);
+    }
+    line = new QFrame(m_gridLayoutWidget);
+    line->setObjectName(QString::fromUtf8("line%1").arg(m_widgetNumber));
```

Comment:
It's not needed to set an object name.

Reply:
Should I remove all instances of setObjectName?

Reply:
Probably, unless you're using those names somewhere, but I don't think so.